lib/stm32/can.c

```diff
@@ -75,8 +77,10 @@ void can_reset(uint32_t canport)
 		rcc_peripheral_reset(&RCC_APB1RSTR, RCC_APB1RSTR_CAN1RST);
 		rcc_peripheral_clear_reset(&RCC_APB1RSTR, RCC_APB1RSTR_CAN1RST);
 	} else {
+#if !defined(STM32F3)
 		rcc_peripheral_reset(&RCC_APB1RSTR, RCC_APB1RSTR_CAN2RST);
 		rcc_peripheral_clear_reset(&RCC_APB1RSTR, RCC_APB1RSTR_CAN2RST);
+#endif
```

Comment:
Can you find a way to do this based on, say, the existance of the CAN2 definition instead?  it will make it more portable going forward for other devices.